relay-server/src/utils/metrics_rate_limits.rs

```diff
@@ -34,6 +36,77 @@ pub struct MetricsLimiter<Q: AsRef<Vec<Quota>> = Vec<Quota>> {
 
 const PROFILE_TAG: &str = "has_profile";
 
+/// Extracts the transaction count from a metric.
+///
+/// If the metric was extracted from a or multiple transaction, it returns the amount
+/// of datapoints contained in the bucket.
+///
+/// Additionally tracks whether the transactions also contained profiling information.
+///
+/// Returns `None` if the metric was not extracted from transactions.
+pub fn extract_transaction_count(
+    metric: &BucketView<'_>,
+    mode: ExtractionMode,
+) -> Option<TransactionCount> {
+    let mri = match MetricResourceIdentifier::parse(metric.name()) {
+        Ok(mri) => mri,
+        Err(_) => {
+            relay_log::error!("invalid MRI: {}", metric.name());
+            return None;
+        }
+    };
+
+    if mri.namespace != MetricNamespace::Transactions {
+        return None;
+    }
+
+    let usage = matches!(mode, ExtractionMode::Usage);
+    let count = match metric.value() {
+        BucketViewValue::Counter(c) if usage && mri.name == "usage" => c as usize,
+        BucketViewValue::Distribution(d) if !usage && mri.name == "duration" => d.len(),
+        _ => 0,
+    };
+
+    let has_profile =
+        matches!(mri.name, "usage" | "duration") && metric.tag(PROFILE_TAG) == Some("true");
+
+    Some(TransactionCount { count, has_profile })
+}
+
+/// Wether to extract transaction and profile count based on the usage
+/// or duration metric.
```

Comment:
```suggestion
/// Wether to extract transaction and profile count based on the usage or duration metric.
```

might as well be one line